c7n/resources/vpc.py

```diff
@@ -18,13 +18,56 @@
 from c7n.manager import resources
 from c7n.utils import local_session, type_schema
 
+from c7n.filters import ValueFilter, Filter, FilterRegistry
+from c7n.utils import local_session, type_schema
+
 
 @resources.register('vpc')
 class Vpc(QueryResourceManager):
 
     resource_type = 'aws.ec2.vpc'
 
 
+
+@Vpc.filter_registry.register('subnets')
+class SubnetsOfVpc(ValueFilter):
+
+    schema = type_schema('subnets', rinherit=ValueFilter.schema)
+
+    def __init__(self, *args, **kw):
+        super(SubnetsOfVpc, self).__init__(*args, **kw)
+        self.data['key'] = 'Subnets'
+
+    def process(self, resources, event=None):
+
+        def _user_vpc_subnets(resource):
```

Comment:
reuse the subnets resource manager instead of inline fetching them to utilize the cache.